os/kernel/debug/sysdbg.c

```diff
@@ -628,8 +803,55 @@ void save_task_scheduling_status(struct tcb_s *tcb)
 #endif
 		sysdbg_struct->sched[index].pid = tcb->pid;
 		sysdbg_struct->sched[index].sched_priority = tcb->sched_priority;
-		sysdbg_struct->sched[index].ptcb = tcb;
+		sysdbg_struct->sched[index].context_switch_reason = CONTEXT_DEFAULT;
```

Comment:
Could you let me know why this is necessary even we have the switch case at below?